### PR TITLE
Add docs and tests

### PR DESCRIPTION
**Pull request by maxsokolovsky**
There are open PRs with changes to PRTB, CRTB, Token and other resource validation and mutation. I will update the docs when they are merged.

### CODE REVIEW

File: pkg/resources/core/v1/secret/mutator_test.go
Comment (JonCrowther) [inline pkg/resources/core/v1/secret/mutator_test.go:414]: Minor, but I believe you could move all this ctrl portion above the for loop since none of it relies on the tests themselves

File: pkg/resources/management.cattle.io/v3/feature/validator_test.go
Comment (JonCrowther) [inline pkg/resources/management.cattle.io/v3/feature/validator_test.go:131]: I'm not particularly fond of having this custom logic for one of the tests. Maybe add the `AdmissionRequest` as one of the test fields?

**Review by JonCrowther — CHANGES_REQUESTED** [review]
Everything looks good, just some suggestions for the structure of the tests

File: pkg/resources/core/v1/secret/mutator_test.go
Comment (maxsokolovsky) [reply pkg/resources/core/v1/secret/mutator_test.go:414]: Sure, I'll move this above the test loop, since each test case here requires the mutator to be constructed in the same way.

File: pkg/resources/management.cattle.io/v3/feature/validator_test.go
Comment (maxsokolovsky) [reply pkg/resources/management.cattle.io/v3/feature/validator_test.go:131]: Agreed, but the setup will be less clear in how the inputs are defined. I am testing features, so my inputs are v3.Feature values. Otherwise, I'd have to define `admissionv1.AdmissionRequest` for each test instead, which would be fine, since the method under test, Admit, actually takes a value of that type. But, again, readability might suffer a bit, since I really care about Features.
It's not ideal, but the test case itself is explicitly declared with the name and expected result.

**Review by JonCrowther — DISMISSED** [review]

**Review by KevinJoiner — CHANGES_REQUESTED** [review]

**Review by KevinJoiner — DISMISSED** [review]
You should be able to update this to merge into the release branch once we get approvals and merge the dev branch into the release. https://github.com/rancher/webhook/pull/256

**Review by KevinJoiner — APPROVED** [review]

**Review by JonCrowther — APPROVED** [review]

**Pull request merged**
